src/containers/AccountConnection.jsx

```diff
@@ -43,6 +43,21 @@ class AccountConnection extends Component {
     this.store.addConnectionStatusListener(konnector, this.connectionListener)
   }
 
+  getFolderPathIfNecessary (connector, account) {
+    const { t } = this.context
+    const folderField = connector && connector.fields && connector.fields.folderPath
+    const auth = account && account.auth
+    if (folderField && !folderField.default && auth && !auth.folderPath) {
```

Comment:
Maybe we can use early returns to avoid long `else if` list.
```javascript
if (a) {
 return b
}

if (c) {
 return d
}

return null
```
What do you think ?